### DISCUSSION TITLE
How to change search algorithm

### DISCUSSION THREAD

**Discussion by MirS0bhan** [davatorium/rofi#2034] [General]
i have a problem with searching. it doesn't match most related stuffs 
and is it possible to that rofi count how many times i open an app and it consider it in showing items ?

**Comment by DaveDavenport** [comment]
by default run/drun put the most (recently) used applications on top.   (see history option). 

Search algorithm can be changed via config file /cli options, see manpage. 

https://davatorium.github.io/rofi/1.7.5/rofi.1/#history-and-sorting

**Reply by DaveDavenport** [reply]
to change matching:
https://davatorium.github.io/rofi/1.7.5/rofi.1/#matching

**Reply by MirS0bhan** [reply]
`rofi -show drun -matching prefix -drun-match-fields name -sort true -show-icons -no-disable-history -sorting-method fzf`

i run this but history did not work.

**Reply by DaveDavenport** [reply]
Hard to say without more information, but it should work.  The unfiltered list should have most recent op top.

The history files are stored in `~/.cache/rofi3.runcache` and `~/.cache/rofi3.druncache`  (depending on your xdg settings).

**Reply by MirS0bhan** [reply]
this is my cache and it doesn't work 

rofi -version
Version: 1.7.5


> cat .cache/rofi3.druncache
19 firefox.desktop
17 org.telegram.desktop.desktop
9 org.flameshot.Flameshot.desktop
1 org.kde.discover.desktop
1 nekoray.desktop
1 org.kde.konsole.desktop
1 dev.zed.Zed.desktop
1 de.schmidhuberj.tubefeeder.desktop
1 qvidcap.desktop
1 pycharm.desktop
1 idea.desktop
1 btop.desktop
0 org.gnome.Ptyxis.desktop
0 xfce4-about.desktop

**Reply by DaveDavenport** [reply]
Tested with rofi 1.7.5, it works for me give above constraint.

**Reply by MirS0bhan** [reply]
what should I do ?
